template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/CmTemplateProcessor.java

```diff
@@ -253,12 +254,13 @@ public GatewayRecommendation recommendGateway() {
 
     @Override
     public AutoscaleRecommendation recommendAutoscale() {
-        Set<String> time = getRecommendationByBlacklist(BlackListedTimeBasedAutoscaleRole.class, true);
-        Set<String> load = getRecommendationByBlacklist(BlackListedLoadBasedAutoscaleRole.class, true);
+        Set<String> time = getRecommendationByBlacklist(BlackListedTimeBasedAutoscaleRole.class, true, List.of());
+        Set<String> load = getRecommendationByBlacklist(BlackListedLoadBasedAutoscaleRole.class, true, List.of());
         return new AutoscaleRecommendation(time, load);
     }
 
-    private <T extends Enum<T>> Set<String> getRecommendationByBlacklist(Class<T> enumClass, boolean emptyServiceListBlacklisted) {
+    private <T extends Enum<T>> Set<String> getRecommendationByBlacklist(Class<T> enumClass, boolean emptyServiceListBlacklisted,
+        List<String> entitlements) {
```

Comment:
I would add some logs here to be able to identify which entitlements and for which enum were parameterezied.